### PR TITLE
Track 10 - basic OAuth

### PR DESCRIPTION
**Pull request by tomca32**
This introduces basic Google Oauth atuhentication. Users can now login with any valid google account. This is currently running on staging so feel free to try it out.

This is using only omniauth gems. No devise yet. I wanted to get this out and running and introduce Devise as a separate task.

@j16r :bowtie:

### CODE REVIEW

**Comment by tomca32** [comment]
Added Devise

**Comment by tomca32** [comment]
Hakiri check fails because of XSS vulnerability in Rails 5. There's a different PR to fix that.

File: README.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ Things you'll need
 Starting the server
 
     bundle install
-    rails s
+    GOOGLE_CLIENT_ID={your_google_client_id} GOOGLE_CLIENT_SECRET={your_google_client_secret} rails s
```
Comment (j16r) [inline README.md]: could we not put this in the config somehow?

File: config/environments/development.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 Rails.application.configure do
   # Settings specified here will take precedence over those in config/application.rb.
-
+  OpenSSL::SSL.send(:remove_const, :VERIFY_PEER)
```
Comment (j16r) [inline config/environments/development.rb]: Is this the only way to disable SSL peer verification? Is that what this is doing?

**Comment by j16r** [comment]
Rebase onto 5.0.0.1 and make sure it works, then merge!

**Pull request merged**